src/components/todo.js

```diff
@@ -0,0 +1,120 @@
+import { $ } from '../utils/dom.js';
+import { todos } from '../../src/models/todos.js'
+import { uncheckedIcon, checkedIcon, deleteIcon } from "../../assets/svgIcon.js";
+
+// input dom
+const $todoInput = $('#todoInput');
+// ul dom
+const $todoList = $('#todoList');
+
+// SVG 아이콘을 생성하는 함수
+const createIcon = (svgPath) => {
+  const buttonIcon = document.createElement('img');
+  buttonIcon.src = `data:image/svg+xml,${encodeURIComponent(svgPath)}`;
+  return buttonIcon;
+};
+
+// 할 일 li를 생성하는 함수 (checkArea, todoText, deleteButton으로 구성)
+const formatLi = (text, checkStatus) => {
+  // 동적으로 생성된 li
+  const todoLi = document.createElement('li');
+  // check영역 생성
+  const checkArea = document.createElement('span');
+  // 입력된 todo 넣을 영역 생성
+  const todoText = document.createElement('span');
+  // deleteButton 생성
+  const deleteButton = createIcon(deleteIcon);
+
+  // 전달 받은 값을 li의 텍스트 영역에 넣음
+  todoText.textContent = text;
+
+  // 상태에 따른 아이콘 설정
+  checkArea.appendChild(createIcon(checkStatus ? checkedIcon : uncheckedIcon));
+
+  // 체크영역을 클릭할 경우
+  checkArea.addEventListener('click', () => {
+    // 일치하는 항목을 탐색
+    const todoItem = todos.find((ele) => ele.text === text && ele.isChecked === checkStatus);
```

Comment:
p3. 아래와 같이 find, map, filter등의 배열 메소드를 사용할 때는 복수형.find(단수형 => {}) 이렇게 네이밍을 가져가는게 좋아요~

```suggestion
    const todoItem = todos.find((todo) => todo.text === text && todo.isChecked === checkStatus);
```